hack/remote-debug/README.md

```diff
@@ -0,0 +1,512 @@
+
+    You need to specify several environment variables for the build and
+    deployment processes. These variables can be set in a `.envrc` file in
+    the top-level directory.
+
+    - **REGISTRY**: The container registry where the image will be pushed.
+
+    - **IMAGE**: The name of the debug image.
+
+    - **TAG**: (Optional, defaults to "latest"): Specifies the tag for the image.
+
+    - **GITHUB_USER**: When building the custom debug image with SSH support,
+      only public key authentication is supported for SSH login to the pod.
+      The public keys are fetched from
+      `https://github.com/${GITHUB_USER}.keys`, and the Dockerfile requires
+      this environment variable to be set.
+
+    - **DOCKERFILE_DEBUG**: This ensures that the Makefile uses the correct
+      Dockerfile for building the debug image, enabling features necessary
+      for remote debugging.
+
+    - **HAPROXY_RPMS**: Specifies the URLs or package names of the HAProxy
+      RPM packages to be used.
+
+    You should verify that you have an SSH key installed on Github with:
+
+    ```sh
+    curl https://github.com/${GITHUB_USER}.keys
+    ```
+
+#### Example `.envrc` file:
+
+```sh
+export REGISTRY=quay.io
+export IMAGE=amcdermo/openshift-router
+export GITHUB_USER=frobware
+export DOCKERFILE_DEBUG=hack/remote-debug/Dockerfile
+export HAPROXY_RPMS="https://github.com/frobware/haproxy-builds/raw/master/debug/rhaos-4.17-rhel-9/haproxy-debugsource-2.8.10-1.rhaos4.17.el9.x86_64.rpm https://github.com/frobware/haproxy-builds/raw/master/debug/rhaos-4.17-rhel-9/haproxy28-2.8.10-1.rhaos4.17.el9.x86_64.rpm https://github.com/frobware/haproxy-builds/raw/master/debug/rhaos-4.17-rhel-9/haproxy28-debuginfo-2.8.10-1.rhaos4.17.el9.x86_64.rpm"
+```
+
+You can either source the `.envrc` file manually with:
+```sh
+$ source .envrc
+```
+or, if using [direnv](https://direnv.net/), the environment variables
+will automatically load when you navigate to the directory.
+
+# Building and deploying the debug image
+
+1. **Build `openshift-router` and `remote-debug-helper`**:
+
+   Navigate to the top-level directory of your
+   [openshift-router](https://github.com/openshift/router) clone and
+   run the following command to compile the `openshift-router` and
+   `remote-debug-helper` binaries:
+
+    ```sh
+    $ make -f hack/Makefile.debug build-image
+    GO111MODULE=on CGO_ENABLED=0 GOFLAGS=-mod=vendor go build -o openshift-router -gcflags=all="-N -l" ./cmd/openshift-router
+    GO111MODULE=on CGO_ENABLED=0 GOFLAGS=-mod=vendor go build -o hack/remote-debug/remote-debug-helper -gcflags=all="-N -l" ./hack/remote-debug/remote-debug-helper.go
+    podman build --build-arg HAPROXY_RPMS=https="https://github.com/frobware/haproxy-builds/raw/master/rhaos-4.17-rhel-9/haproxy28-2.8.10-1.rhaos4.17.el9.x86_64.rpm" --build-arg GITHUB_USER="frobware" -t amcdermo/openshift-router:latest -f hack/Dockerfile.debug .
+    ...
+    Successfully tagged localhost/amcdermo/openshift-router:latest
```

Comment:
But this is just the output from podman build. The push target uses the REGISTRY value.

```Makefile
push-image:
	$(IMAGEBUILDER) tag $(IMAGE):$(TAG) $(REGISTRY)/$(IMAGE):$(TAG)
	$(IMAGEBUILDER) push $(REGISTRY)/$(IMAGE):$(TAG)
```

Reply:
Yes, I just wanted to say that the output doesn't look like you run all the commands you suggest the user to run. Because if the user would do as you showed in `.envrc` example it would be `Successfully tagged quay.io/amcdermo/openshift-router:latest`. Probably nobody would even notice that though.

Reply:
Sorry my bad, I forgot the this command doesn't tag with the image registry, it's done at the push time.